### PR TITLE
Update d3d12 SDK 10.0.14393.0 Remove d3d12-sys

### PR DESCRIPTION
**Pull request by red75prime**

### CODE REVIEW

File: src/um/d3d12.rs [hunk 1]
```diff
@@ -1638,260 +1792,606 @@ STRUCT!{struct D3D12_STATIC_SAMPLER_DESC {
...
+
+RIDL!{#[uuid(0x5b160d0f, 0xac1b, 0x4185, 0x8b, 0xa8, 0xb3, 0xae, 0x42, 0xa5, 0xa4, 0x55)]
+interface ID3D12GraphicsCommandList(ID3D12GraphicsCommandListVtbl): 
+    ID3D12CommandList(ID3D12CommandListVtbl) {
+    fn Close() -> HRESULT,
+    fn Reset(
+        pAllocator: *mut ID3D12CommandAllocator,
+        pInitialState: *mut ID3D12PipelineState,
+    ) -> HRESULT,
+    fn ClearState(pPipelineState: *mut ID3D12PipelineState,) -> (),
```
Comment (ekicyou) [inline src/um/d3d12.rs]: *mut --> *const ?
d3d12.idl ,[annotation("_In_")]  or [annotation("_In_opt_")] com pointers

Comment (red75prime) [inline src/um/d3d12.rs]: Semantically it should me *mut. 'Reset' must be able to change data hidden behind interface pointer. Also, there are pointer-to-const-data parameters, annotated as _In_ too.  Preserving this distinction is safe bet I guess.

Comment (retep998) [inline src/um/d3d12.rs]: There's no const in the header, therefore `*mut` is correct.

File: src/um/d3d12.rs [hunk 2]
```diff
@@ -1,25 +1,27 @@
-// Copyright © 2015-2017 winapi-rs developers
+// Copyright © 2017 winapi-rs developers
```
Comment (retep998) [inline src/um/d3d12.rs]: Don't update the start year of the copyrights.

File: src/um/d3d12.rs [hunk 3]
```diff
@@ -422,36 +416,42 @@ pub const D3D12_WHQL_DRAWINDEXED_INDEX_COUNT_2_TO_EXP: UINT = 25;
 pub const D3D12_WHQL_DRAW_VERTEX_COUNT_2_TO_EXP: UINT = 25;
 pub type D3D12_GPU_VIRTUAL_ADDRESS = UINT64;
 ENUM!{enum D3D12_COMMAND_LIST_TYPE {
-    D3D12_COMMAND_LIST_TYPE_DIRECT = 0,
-    D3D12_COMMAND_LIST_TYPE_BUNDLE = 1,
-    D3D12_COMMAND_LIST_TYPE_COMPUTE = 2,
-    D3D12_COMMAND_LIST_TYPE_COPY = 3,
+    D3D12_COMMAND_LIST_TYPE_DIRECT = 0x0, // 0
```
Comment (retep998) [inline src/um/d3d12.rs]: Why aren't you preserving the way the number was formatted in the header?

File: src/um/d3d12.rs [hunk 4]
```diff
@@ -622,64 +640,73 @@ STRUCT!{struct D3D12_RASTERIZER_DESC {
     ForcedSampleCount: UINT,
     ConservativeRaster: D3D12_CONSERVATIVE_RASTERIZATION_MODE,
 }}
+
 RIDL!{#[uuid(0xc4fec28f, 0x7966, 0x4e95, 0x9f, 0x94, 0xf4, 0x31, 0xcb, 0x56, 0xc3, 0xb8)]
 interface ID3D12Object(ID3D12ObjectVtbl): IUnknown(IUnknownVtbl) {
     fn GetPrivateData(
-        guid: REFGUID,
```
Comment (retep998) [inline src/um/d3d12.rs]: Despite `REFGUID` being a macro and not a proper typedef in the original headers, I treat it as a proper typedef in winapi.

File: src/um/d3d12.rs [hunk 5]
```diff
@@ -622,64 +640,73 @@ STRUCT!{struct D3D12_RASTERIZER_DESC {
...
     ) -> HRESULT,
     fn SetPrivateDataInterface(
-        guid: REFGUID,
+        guid: *const GUID,
         pData: *const IUnknown,
     ) -> HRESULT,
-    fn SetName(
-        Name: LPCWSTR,
-    ) -> HRESULT,
+    fn SetName(Name: LPCWSTR,) -> HRESULT,
```
Comment (retep998) [inline src/um/d3d12.rs]: Even with a single parameter, the function should formatted as a block.

File: src/um/d3d12.rs [hunk 6]
```diff
@@ -622,64 +640,73 @@ STRUCT!{struct D3D12_RASTERIZER_DESC {
...
         ppvDevice: *mut *mut c_void,
     ) -> HRESULT,
 }}
+
 RIDL!{#[uuid(0xc54a6b66, 0x72df, 0x4ee8, 0x8b, 0xe5, 0xa9, 0x46, 0xa1, 0x42, 0x92, 0x14)]
-interface ID3D12RootSignature(ID3D12RootSignatureVtbl):
-    ID3D12DeviceChild(ID3D12DeviceChildVtbl) {}}
+interface ID3D12RootSignature(ID3D12RootSignatureVtbl): 
+    ID3D12DeviceChild(ID3D12DeviceChildVtbl) {
+}}
```
Comment (retep998) [inline src/um/d3d12.rs]: Keep the closing brace on the same line as the opening brace if it is empty.

**Comment by docbrown** [comment]
@red75prime ping

**Comment by red75prime** [comment]
@docbrown I'm in a process of solving some sleep-related problems, so I'm not sure when I'll be able to update my code generator to fix the issues.

**Comment by Susurrus** [comment]
I believe this can be closed now that #511 was merged.

@red75prime I hope you got your health issues sorted and thanks for the PR anyways!

**Pull request closed without merging**